extensions/zmh-program/chatgpt-en.js

```diff
@@ -0,0 +1,161 @@
+(function(Scratch) {
+  'use strict';
+  let cache = {};
+  let conf = {
+    apikey: '',
+    model: '',
+    temp: 70,
+    token: 128,
+  };
+
+  const completions = [
+      'text-davinci-003',
+      'text-davinci-002',
+      'text-curie-001',
+      'text-babbage-001',
+      'text-ada-001'
+    ];
+  const chat_completions = [
+      'gpt-3.5-turbo-0301',
+      'gpt-3.5-turbo',
+      'gpt-4-32k-0314',
+      'gpt-4-32k',
+      'gpt-4-0314',
+      'gpt-4'
+    ];
+  const all_completions = [...completions, ...chat_completions];
+  async function request_completions() {
+    // see https://platform.openai.com/docs/api-reference/completions/create
+    // eslint-disable-next-line no-restricted-syntax
+    const resp = await fetch(`https://api.openai.com/v1/completions`, {
```

Comment:
again please do not disable the ESLint warnings without reading them...

Reply:
> Hmm, you say you have redis caching. Doesn't that mean the temperature is effectively 0?

Sorry, I don't understand what you mean. temperature is custom. The default is 0.7.

Reply:
OK